### DISCUSSION TITLE
Hidden column headers still show up when toggled off.

### DISCUSSION THREAD

**Discussion by akulmehta** [Power-Components/livewire-powergrid#1791] [BUG]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

No

### PHP Version

8.3

### PowerGrid

6.1.6

### Laravel

11.34.2

### Livewire

3.5.13

### Alpine JS

3.14.3

### Theme

Tailwind 3.x

### Describe the bug.

I have hidden the ID column using: 

```php
            Column::make('ID', 'id')
                ->searchable()
                ->sortable()
                ->hidden(isHidden: true, isForceHidden: false),
```

However, while the column doesn't show up, the header row still shows the "ID" column but the remaining rows do not causing the columns to be offset. Example:

![image](https://github.com/user-attachments/assets/3cd5401c-bc84-4493-a9b3-57fbc2e36697)

### To Reproduce...

- Create a PowerGrid table. 
- Make a column hidden but toggleable using: 

```php
            Column::make('ID', 'id')
                ->searchable()
                ->sortable()
                ->hidden(isHidden: true, isForceHidden: false),
```

- observe the extra column header.

**Comment by akulmehta** [comment]
Update: Note this only happens when views are published and I believe it is due to a missing semi colon `;` after `display:none` in cols.php `resources/views/components/cols.blade.php` lines 47-52

Specifically the following:

```php
    @style([
        'display:none' => data_get($column, 'hidden') === true,
        'cursor:pointer' => data_get($column, 'enableSort'),
        data_get($column, 'headerStyle') => filled(data_get($column, 'headerStyle')),
        'width: max-content !important',
    ])
```

**Comment by eafarooqi** [comment]
Hi, 
please update to the latest version. this has been fixed in the latest version and republish the views if you have done any custom changes. Please take a backup of your old changes first, otherwise they may get replaced with the latest published views.

**Comment by akulmehta** [comment]
@eafarooqi I did. As mentioned above I am on the latest version. 🤷‍♂️

**Comment by luanfreitasdev** [comment]
The style directive already adds ";" when assembling the style. This could be some cache or published files without updates

**Reply by akulmehta** [reply]
Yeah I thought so as well. Let me see if I can dig more into this issue. I did clear my cache and deleted and republished my views too but that didn't help.

**Comment by luanfreitasdev** [comment]
### Note

From now on we will follow everything in one place. We will use the discussions tab to allow other users to also interact similarly to Livewire, which is why I am turning this issue into a discussion.

Thanks for understanding
